src/inspector_agent.cc

```diff
@@ -706,6 +751,52 @@ void Url(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), url.c_str()));
 }
 
+static void invokeAsyncTaskFnWithId(void (Agent::*asyncTaskFn)(void*),
+                                    const FunctionCallbackInfo<Value>& args);
+
+static void AsyncTaskScheduledWrapper(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsString())
+    return env->ThrowTypeError("taskName must be a string");
+  if (!args[1]->IsNumber())
+    return env->ThrowTypeError("taskId must be a number");
+  if (!args[2]->IsBoolean())
+    return env->ThrowTypeError("recurring must be a boolean");
+
+  Local<String> taskName = args[0].As<String>();
+  String::Value taskNameValue(taskName);
+  StringView taskNameView(*taskNameValue, taskNameValue.length());
+
+  intptr_t task_id = args[1].As<Integer>()->Value();
+  void* task = reinterpret_cast<void*>(task_id);
+  bool recurring = args[2].As<Boolean>()->Value();
+
+  env->inspector_agent()->AsyncTaskScheduled(taskNameView, task, recurring);
+}
+
+static void AsyncTaskCanceledWrapper(const FunctionCallbackInfo<Value>& args) {
+  invokeAsyncTaskFnWithId(&Agent::AsyncTaskCanceled, args);
+}
+
+static void AsyncTaskStartedWrapper(const FunctionCallbackInfo<Value>& args) {
+  invokeAsyncTaskFnWithId(&Agent::AsyncTaskStarted, args);
+}
+
+static void AsyncTaskFinishedWrapper(const FunctionCallbackInfo<Value>& args) {
+  invokeAsyncTaskFnWithId(&Agent::AsyncTaskFinished, args);
+}
+
+static void invokeAsyncTaskFnWithId(void (Agent::*asyncTaskFn)(void*),
+                                    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsNumber())
+    return env->ThrowTypeError("taskId must be a number");
+
+  intptr_t task_id = args[0].As<Integer>()->Value();
```

Comment:
Will do, I was already thinking about this in https://github.com/nodejs/node/pull/13870#discussion_r123700732.